Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Update.ts

```diff
@@ -254,64 +255,87 @@ async function checkForInvalidReferences(
   };
 }
 
+async function updateDocumentByIdTransaction(
+  updateRequest: UpdateRequest,
```

Comment:
I extracted the transaction behavior into a separate function that mirrors the structure of upsert and delete, to simplify the code wrapped in the retry block.